test/simulation.jl

```diff
@@ -412,17 +428,32 @@ end
   end
   @test 4 == length(checkForContractionInGensim(single_contract))
 
-  sim = eval(gensim(contract_with_op2))
-  f = sim(earth, default_dec_generate)
-  A = 3 * ones(nv(earth))
-  E_dec = ones(nv(earth))
-  u = ComponentArray(A=A, E=E_dec)
-  du = ComponentArray(A=zeros(ntriangles(earth)), E=zeros(nv(earth)))
-  constants_and_parameters = ()
-  f(du, u, constants_and_parameters, 0)
+  let sim = eval(gensim(contract_with_op2))
```

Comment:
Find some way eliminating duplicated code. Loop over `[true,false]` for the value of the `can_prealloc` flag.